### PR TITLE
Fix/rock collide animation

### PR DESCRIPTION
**Pull request by mitchellolsthoorn**

### CODE REVIEW

File: desktop/src/main/nl/tudelft/ti2806/riverrush/controller/RenderController.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public RenderController(final Game gm, final EventDispatcher eventDispatcher) {
         this.dispatcher.attach(GameFinishedEvent.class, this.onGameFinishedLambda);
         this.dispatcher.attach(AssetsLoadedEvent.class, this.onAssetsLoadedLambda);
         this.dispatcher.attach(AnimalCollidedEvent.class, this.sendOverNetworkLambda);
-        this.dispatcher.attach(BoatCollidedEvent.class, this.sendOverNetworkLambda);
+        //this.dispatcher.attach(BoatCollidedEvent.class, this.sendOverNetworkLambda);
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/controller/RenderController.java]: Just remove this

File: desktop/src/main/nl/tudelft/ti2806/riverrush/desktop/MainDesktop.java [hunk 2]
```diff
@@ -53,6 +59,52 @@ public MainDesktop() throws URISyntaxException, InterruptedException {
...
+        Thread.sleep(5000);
+        event.setLocation(Direction.NEUTRAL);
+        event.setTeam(0);
+        this.eventDispatcher.dispatch(event);
+
+        Thread.sleep(5000);
+        event.setLocation(Direction.NEUTRAL);
+        event.setTeam(1);
+        this.eventDispatcher.dispatch(event);
     }
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/desktop/MainDesktop.java]: This is temporary code I assume, please remove this before merging

File: desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/RockGraphic.java [hunk 3]
```diff
@@ -79,6 +85,9 @@ public void draw(final Batch batch, final float parentAlpha) {
         TextureRegion region = new TextureRegion(tex, 0, 0, TEXTURE_SIZE_X, TEXTURE_SIZE_Y);
         batch.enableBlending();
 
+        Color color = this.getColor();
+        batch.setColor(color.r, color.g, color.b, color.a * parentAlpha);
+
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/RockGraphic.java]: I assume the fade is not working yet? A possible solution is resetting the color after the batch.draw call (see some of the other graphics classes for example)

File: desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/RockGraphic.java [hunk 4]
```diff
@@ -105,4 +115,35 @@ public Direction getDirection() {
...
+        final float shakeSpeed = 0.02f;
+
+        this.addAction(
+            sequence(
+            parallel(
+                Actions.scaleTo(0, 0, 1f),
+                Actions.moveBy(-800, 0, 1f),
+                Actions.repeat(shakeCount, (
+                    Actions.sequence(
+                        Actions.moveBy(shakeDistance, 0, shakeSpeed),
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/RockGraphic.java]: You can make this a slightly cleaner call using the repeat Action.

File: desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/RockGraphic.java [hunk 5]
```diff
@@ -105,4 +115,35 @@ public Direction getDirection() {
...
+    /**
+     * Display the collide animation on the rock
+     * and then remove the rock from the game.
+     */
+    public void collideAnimation() {
+        final int shakeCount = 10;
+        final int shakeDistance = 20;
+        final float shakeSpeed = 0.02f;
+
+        this.addAction(
```
Comment (JobEngel) [inline desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/RockGraphic.java]: I'd personally recommend separating these calls

**Comment by txsmith** [comment]
Why did you open a PR for this? This code is not PR ready.

**Comment by txsmith** [comment]
Still needs work.

**Pull request closed without merging**